posixlib/src/main/resources/scala-native/pthread.c

```diff
@@ -49,6 +33,7 @@ int scalanative_pthread_mutex_normal() { return PTHREAD_MUTEX_NORMAL; }
 int scalanative_pthread_mutex_recursive() { return PTHREAD_MUTEX_RECURSIVE; }
 
 pthread_once_t scalanative_pthread_once_init() {
+    // On macOS, PTHREAD_ONCE_INIT is defined as an expression
```

Comment:
So that is the reason for the apparently convoluted implementation.
I thank you and future readers will probably also thank you,
particularly when they make to 'obvious` change on Linux and
it blows up on macOS.

I would not change the comment, but for future readers tracing evolution:
I believe the issue on macOS is not that it is defined as a (simple) 
expression but rather that it is defined as a block {}.  The comment
as is is __far__ better than what was before and gives at least a clue
to anybody unfortunate enough to be in this method.